src/lib/scheduler/shutdown_task.cpp

```diff
@@ -0,0 +1,10 @@
+#include "shutdown_task.hpp"
+
+namespace hyrise {
+
+void ShutdownTask::_on_execute() {
+  Assert(_active_worker_count.load() > 0, "Cannot have less than zero active workers.");
```

Comment:
Use the returned value of `--active_worker_count` and assert after decrementing to get the value in one atomic operation? Maybe use post decrement to get the value before decrementing.

Reply:
I think the current version is simpler and it's solely used when shutting down the scheduler. So accessing the atomic twice shouldn't make a difference.

Or is it about something else?

Reply:
Not about performance, but in theory the Assert could not be triggered with the current code if too many tasks decrement the value.

Reply:
(I could imagine a test for this...)